src/base/QXmppMessageReaction.cpp

```diff
@@ -0,0 +1,121 @@
+// SPDX-FileCopyrightText: 2022 Melvin Keskin <melvo@olomono.de>
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "QXmppMessageReaction.h"
+
+#include "QXmppConstants_p.h"
+#include "QXmppUtils.h"
+
+#include <QDomElement>
+
+class QXmppMessageReactionPrivate : public QSharedData
+{
+public:
+    QString id;
+    QVector<QString> emojis;
+};
+
+///
+/// \class QXmppMessageReaction
+///
+/// \brief The QXmppMessageReaction class represents a reaction to a message in the form of emojis
+/// as specified by \xep{0444: Message Reactions}.
+///
+/// \since QXmpp 1.5
+///
+
+///
+/// Constructs a message reaction.
+///
+QXmppMessageReaction::QXmppMessageReaction()
+    : d(new QXmppMessageReactionPrivate)
+{
+}
+
+QXMPP_PRIVATE_DEFINE_RULE_OF_SIX(QXmppMessageReaction)
+
+///
+/// Returns the ID of the message for that the reaction is sent.
+///
+/// \return the message's ID
+///
```

Comment:
please explain which message id is meant here (there're the normal `id`, origin id, stanza id of the server, and maybe something i've missing)